fix(core): make `menuCollapsed = true` on small screen

Richard-Mackey opened the pull request:
### ✨ Description
This PR fixes the sidebar behaviour on mobile devices to improve UX on small screens.

**Changes:**
- Sidebar now defaults to collapsed state on mobile viewports
- Desktop behaviour unchanged - respects user's localStorage preference
- Added comments explaining the mobile-first initialisation logic

**Implementation:**
- Check screen size on component initialisation using `useMediaQuery`
- Set initial collapsed state based on viewport width
- Maintain existing localStorage functionality for desktop users

### 🔗 Related Issue
Fixes #12725

### 🎨 Frontend Checklist
- [x] Code builds without errors (`npm run build`)
- [ ] E2E tests: 5 failures appear unrelated to sidebar changes (execution bulk actions, flow creation). All failures are in areas not touched by this PR. 
- [x] Screenshots attached showing the UI changes

**Mobile. Default collapsed state:**

<img width="1440" height="753" alt="mobile_default_view" src="https://github.com/user-attachments/assets/b505f8ca-20cc-4d0e-a3b1-fdfb28a72fd3" />

**Mobile. Expanded state:**

<img width="1440" height="753" alt="mobile_expanded_view" src="https://github.com/user-attachments/assets/7ed5a7bc-94fc-4868-af31-97f7b9d58107" />

Richard-Mackey committed:
fix: ensure mobile sidebar defaults to collapsed state on initial load

- Sidebar completely hidden by default on mobile viewports (≤768px)
- Toggle button always visible in top header on mobile
- Clicking toggle shows sidebar as full-screen overlay
- Desktop behavior unchanged - respects localStorage preference
- Added clear comments explaining mobile-first initialisation logic

Fixes #12725

MilosPaunovic reviewed (changes requested):
Hey @Richard-Mackey,

Super cool progress, but what we wanted on mobile screens is to sidebar be completely collapsed by default, like when you click the button in the top header.

Can you amend your PR to handle that?

Richard-Mackey commented:
Updated to fully hide the sidebar on mobile (not just icon strip).

**Mobile default view with sidebar completely hidden:**
<img width="347" height="602" alt="mobile_view_collapsed_sidebar" src="https://github.com/user-attachments/assets/fb1125ab-a810-4ad6-9c55-43f63b061c96" />

MilosPaunovic commented:
We still need to have the button in header to show it if we want to.

MilosPaunovic commented:
Hey @Richard-Mackey, are there any updates on this?

Richard-Mackey commented:
Hey @MilosPaunovic, I ran out of time on Friday. I will have a fix for it today.

Richard-Mackey commented:
Hey @MilosPaunovic, I've updated the PR: the sidebar should now be completely hidden by default on mobile, with the toggle button still in the top header to show/hide it 👍

MilosPaunovic reviewed (changes requested):
Hello @Richard-Mackey,

This is the current state now, on large screens:

<img width="1200" height="592" alt="image" src="https://github.com/user-attachments/assets/fb399c47-338c-4c48-b1e8-54d054fc6fb9" />

We do not want that.

What we're trying to achieve is the following:

1. On small screens, detected, automatically close the menu, but show the header toggle button to be able to expand it, no mini version.
2. On large screen, automatically show the menu, but show the header toggle button to be able to close it, no mini version.

With your changes made to support this, you'll need to upload tow screenshots, one of how it looks on small screens and one on a large screen (a screenshare video is also okay).

Richard-Mackey committed:
fix: remove mini icon sidebar on both mobile and desktop

- Desktop: sidebar now completely hidden when collapsed (not icon-only)
- Mobile: sidebar remains completely hidden when collapsed
- Both viewports: no mini icon version as requested

Richard-Mackey commented:
Thanks for the clarification - I'd misunderstood the requirement initially. I've now updated the PR to ensure there's no mini icon version on either mobile or desktop. The sidebar now:

- Mobile: defaults to hidden, can be toggled to overlay
- Desktop: defaults to expanded, can be toggled to completely hidden

Screenshots showing the behaviour:
Mobile view on opening:
<img width="332" height="589" alt="Mobile-on-open" src="https://github.com/user-attachments/assets/f939257d-7d48-4ef3-829a-4ea6635214ec" />
Mobile view with sidebar open:
<img width="332" height="589" alt="Mobile-sidebar" src="https://github.com/user-attachments/assets/d5ccd17d-0036-4a39-ad3a-06d22a996bc3" />
Desktop view on opening:
<img width="1433" height="659" alt="Desktop-on-open" src="https://github.com/user-attachments/assets/efa4a2ba-8407-43df-b333-8935a350af97" />
Desktop view with sidebar collapsed:
<img width="1433" height="659" alt="Desktop-sidebar-collapse" src="https://github.com/user-attachments/assets/0a356279-c146-4e7c-98df-5a783ed4d9d9" />

Let me know if anything else needs changing 👍

MilosPaunovic commented:
@Piyush-r-bhaskar Could you give this a review once you get the chance? 🙏🏻

Piyush-r-bhaskar reviewed (changes requested):
Hey @Richard-Mackey 

Thanks for your PR 🚀

I see the Issue expectation is : `On low resolution the menu should be close by default`. and it works

But the changes in PR, i mean the styles  and it's title is very confusing. Could you just check how we play with `menuCollapsed` key in local storage and just make your changes as we need from issue and also please remove the unncessary comments before I start to take a look again.

Piyush-r-bhaskar committed:
fix(core): make `menuCollapsed = true` on small screen

Piyush-r-bhaskar reviewed (approved):
Please Ignore my comment.

Thank you for your contribution 🚀

Richard-Mackey committed:
fix: sidebar defaults to collapsed on mobile screens

- Mobile: sidebar hidden by default on page load (≤768px viewports)
- Mobile: toggle button shows sidebar as full-screen overlay when opened
- Desktop: maintains original behaviour (conditional rendering)
- Closes #12725

Piyush-r-bhaskar committed:
Revert "fix: sidebar defaults to collapsed on mobile screens"

This reverts commit 49adeb7c68fa58aed043fbac50b2fbc48e029aaf.

Richard-Mackey commented:
I've updated the PR based on your feedback. Changes made:

- Removed long comments from the code
- Simplified the implementation to focus only on the mobile default behaviour from issue #12725
- Desktop behaviour now uses the original conditional rendering pattern (no CSS patches)
- Mobile uses CSS overlay approach while keeping component in DOM for toggle functionality

**Testing:**
-  Mobile: Sidebar defaults to hidden on page load
-  Mobile: Toggle button shows sidebar as full-screen overlay
- Desktop: Sidebar maintains original hide/show behaviour via conditional rendering

Screenshots attached:
Mobile on load
<img width="368" height="586" alt="mobile-on-open" src="https://github.com/user-attachments/assets/84cbfce6-b6ac-4c44-bef9-91b877e1e218" />
Mobile with sidebar
<img width="368" height="586" alt="mobile-sidebar" src="https://github.com/user-attachments/assets/a3d7f507-aa4d-402a-a374-2198e185a567" />
Desktop on open
<img width="1429" height="654" alt="desktop-on-open" src="https://github.com/user-attachments/assets/6f92f612-13f9-498e-8e4c-5846a0c417ef" />
Desktop sidebar closed
<img width="1429" height="654" alt="desktop-sidebar-closed" src="https://github.com/user-attachments/assets/3c7ca51a-fbe0-411e-943a-caab24e54e94" />

Piyush-r-bhaskar committed:
fix(core): make `menuCollapsed = true` on small screen
